lib/puppet/provider/graylog_api/graylog_api.rb

```diff
@@ -21,21 +21,38 @@ def self.prefetch(resources)
     password = resources[:api][:password]
     port = resources[:api][:port]
     username = resources[:api][:username]
+    tls = resources[:api][:tls]
+    server = resources[:api][:server]
+    verify_tls = resources[:api][:verify_tls]
+    ssl_ca_file = resources[:api][:ssl_ca_file]
     Puppet::Provider::GraylogAPI.api_password = password
     Puppet::Provider::GraylogAPI.api_port = port
-    Puppet::Provider::GraylogAPI.api_username = username
-    wait_for_api(port)
-    resources[:api].provider = new({password: password, port: port, username: username})
+    Puppet::Provider::GraylogAPI.api_tls = tls
+    Puppet::Provider::GraylogAPI.api_server = server
+    Puppet::Provider::GraylogAPI.verify_tls = verify_tls
+    Puppet::Provider::GraylogAPI.ssl_ca_file = ssl_ca_file
+
+    # Settings needed before connection to the server.  We set them on the global
+    # level, so we only need to define them once here.
+    if tls
+      HTTParty::Basement.default_options.update(verify: verify_tls)
+      # we configure this anyway, even we don't want to verify the certificate
+      HTTParty::Basement.default_options.update(ssl_ca_file: ssl_ca_file)
+    end
+    wait_for_api(port, server)
+    wait_for_api(port, server)
```

Comment:
Is there a reason you call `wait_for_api` twice in rapid succession, here?

Reply:
No reason at all.  Think I yanked one line too much .. will remove one